app/services/order_purchaser.rb

```diff
@@ -79,12 +79,19 @@ def quantities
     order.order_details.order("order_details.id").pluck(:quantity)
   end
 
-  def send_facility_notification?
-    order_notification_recipient.present? && !acting_as?
-  end
+  def send_purchase_notifications
+    if !acting_as? || params[:send_notification] == "1"
```

Comment:
how about just locally assigned variables? I prefer not to add methods unless they actually need to be called more than just one place